File: 0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int cs=0;
        int ms=Integer.MIN_VALUE;
        
        for(int i=0;i<nums.length;i++){
            int sum=cs+nums[i];
            if(sum<nums[i]){
                cs=nums[i];
            } 
            else{
                cs=sum;
            }
            if(ms<cs){
                ms=cs;
            }
        }
        return ms;
    }
}

File: 0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        k_max=nums[0]
        sum1=nums[0] 
        for i in range(1,len(nums)):
            sum1=max(nums[i],sum1+nums[i]) 
            k_max=max(sum1,k_max)
        return k_max
